aptos-move/block-executor/src/scheduler.rs

```diff
@@ -254,7 +255,12 @@ impl Scheduler {
                             // Upgrade the execution status read lock to write lock.
                             // Can commit.
                             *status_write = ExecutionStatus::Committed(incarnation);
+
                             *commit_idx += 1;
+                            if *commit_idx == self.num_txns {
+                                // All txns have been committed, the parallel execution can finish.
+                                self.done_marker.store(true, Ordering::SeqCst);
+                            }
```

Comment:
nit: can return None here

Reply:
We do need to change the caller logic that relies on returning None, or have some sort of check in the beginning, else we will hit out of bounds error in try_commit.
Adjusted. The invariant now is that we should never call try_commit after everything is committed, lol